pkg/repro/repro.go

```diff
@@ -421,13 +435,18 @@ func (ctx *context) simplifyProg(res *Result) (*Result, error) {
 		ctx.stats.SimplifyProgTime = time.Since(start)
 	}()
 
-	opts := res.Opts
-	opts.Collide = false
-	crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
-	if err != nil {
-		return res, err
+	if res.ops.Collide {
+		opts := res.Opts
+		opts.Collide = false
+		crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
+		if err != nil {
+			return res, err
+		}
+		if crashed {
+			res.Opts = opts
+		}
 	}
-	if crashed {
+	if !res.ops.Collide && res.ops.Threaded {
 		res.Opts = opts
 		opts.Threaded = false
 		crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
```

Comment:
This look like an overkill for reflection. Normal code will do:

```
type Simplify func(opts *csource.Options) bool

simplifies := []Simplify{
	func(opts *csource.Options) bool {
		if !opts.Collide {
			return false
		}
		opts.Collide = false
		return true
	},
	... other simplications ...
}
```

https://go-proverbs.github.io/